ed25519_utils.go

```diff
@@ -0,0 +1,53 @@
+package jwt
+
+import (
+	"crypto"
+	"encoding/asn1"
+	"encoding/pem"
+
+	"golang.org/x/crypto/ed25519"
+)
+
+type OI struct {
+	ObjectIdentifier asn1.ObjectIdentifier
+}
+
+type ed25519PrivKey struct {
+	Version          int
+	ObjectIdentifier OI
+	PrivateKey       []byte
+}
+
+type ed25519PubKey struct {
+	OBjectIdentifier OI
+	PublicKey        asn1.BitString
+}
+
+// Parse PEM-encoded Edwards curve private key
+func ParseEdPrivateKeyFromPEM(key []byte) (crypto.PrivateKey, error) {
+	var block *pem.Block
+	if block, _ = pem.Decode(key); block == nil {
+		return nil, ErrKeyMustBePEMEncoded
+	}
+	var asn1PrivKey ed25519PrivKey
+	if _, err := asn1.Unmarshal(block.Bytes, &asn1PrivKey); err != nil {
```

Comment:
You could use something along these lines instead:

```
// Parse the key
var parsedKey interface{}
if parsedKey, err = x509.ParsePKCS8PrivateKey(block.Bytes); err != nil {
	return nil, err
}

var pkey ed25519.PrivateKey
var ok bool
if pkey, ok = parsedKey.(ed25519.PrivateKey); !ok {
	return nil, ErrNotEdPrivateKey
}

return pkey
```

This would avoid introducing new temporary structs.